classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+		self::api_call( 'check_in', $data );
+	}
+
+	/**
+	 * Makes HTTP request to our API endpoint
+	 *
+	 * @param string $action The specific endpoint in our API.
+	 * @param array $data Any data to send in the body.
+	 * @return array|bool False if WP Error. Otherwise, array response from wp_remote_post.
+	 * @since 1.11.0
+	 */
+	public static function api_call( $action = '', $data = array() ) {
+		$response = wp_remote_post( 'https://api.wppopupmaker.com/wp-json/pmapi/v2/' . $action, array(
+			'method'      => 'POST',
+			'timeout'     => 20,
+			'redirection' => 5,
+			'httpversion' => '1.1',
+			'blocking'    => false,
+			'body'        => $data,
+			'user-agent'  => 'POPMAKE/' . Popup_Maker::$VER . '; ' . get_site_url(),
+		));
+
+		if ( is_wp_error( $response ) ) {
+			$error_message = $response->get_error_message();
+			PUM_Utils_Logging::instance()->log( sprintf( 'Cannot send telemetry data. Error received was: %s', esc_html( $error_message ) ) );
+			return false;
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Adds admin notice if we haven't asked before.
+	 *
+	 * @param array $alerts The alerts currently in the alert system.
+	 * @return array Alerts for the alert system.
+	 * @since 1.11.0
+	 */
+	public static function optin_alert( $alerts ) {
+		if ( ! self::should_show_alert() ) {
+			return $alerts;
+		}
+
+		$optin_url = add_query_arg( 'pum_optin_check', 'optin' );
+
+		ob_start();
+		?>
+		<ul>
+			<li><a href="<?php echo esc_attr( $optin_url ); ?>"><strong><?php esc_html_e( 'Allow', 'popup-maker' ); ?></strong></a></li>
+			<li><a href="#" class="pum-dismiss"><?php esc_html_e( 'Do not allow', 'popup-maker' ); ?></a></li>
+			<li><a href="https://docs.wppopupmaker.com/article/528-the-data-the-popup-maker-plugin-collects" target="_blank" rel="noreferrer noopener"><?php esc_html_e( 'Learn more', 'popup-maker' ); ?></a></li>
+		</ul>
+		<?php
+		$html = ob_get_clean();
+		$alerts[] = array(
+			'code'        => 'pum_telemetry_notice',
+			'type'        => 'info',
+			'message'     => esc_html__( "Allow Popup Maker to track this plugin's usage and help us make this plugin better? No user data is sent to our servers. No sensitive data is tracked.", 'popup-maker' ),
+			'html'        => $html,
+			'priority'    => 10,
+			'dismissible' => true,
+			'global'      => false,
+		);
+		return $alerts;
+	}
+
+	/**
+	 * Checks if any options have been clicked from admin notices.
+	 *
+	 * @since 1.11.0
+	 */
+	public static function optin_alert_check() {
+		if ( isset( $_GET['pum_optin_check'] ) ) {
+			if ( 'optin' === $_GET['pum_optin_check'] ) {
+				pum_update_option( 'telemetry', true );
+			}
+		}
+	}
+
+	/**
+	 * Whether or not we should show optin alert
+	 *
+	 * @since 1.11.0
+	 * @return bool True if alert should be shown
+	 */
+	public static function should_show_alert() {
+		return false === self::has_opted_in() && current_user_can( 'manage_options' ) && false == get_option( '_pum_telemetry_notice_dismissed', false );
```

Comment:
IDK if we have one somewhere yet or not but we need to add this to a list of all options we store for the deletion/uninstall routines. Just make a note to delete it as well.

Reply:
Also is this option set by the alert system? I thought it stored dismissals in a single option key. If this isn't stored by that alert dismissal system, then I don't see where you are setting that option.

Also, shouldn't it be `false ===`?

Reply:
Hmmm....I believe that option is generated when the alert is dismissed. Let me test that again and make sure. The alert was properly dismissing and staying gone with this code.

Reply:
Yes, that should be `===`. Updating now.

Reply:
Updated the `==` to `===`.

Reply:
@danieliser Yeah, that option shouldn't be there. I had copied the code from the asset cache alert I wrote where I used an option like that as the flag but I must have kept this on accident here.

It is never set while using this plugin on my local site. What controls the alert from showing again to the user is the alert system and not this option.

Reply:
I believe there is a _pum_dismissed_alerts option saved in user meta: https://github.com/PopupMaker/Popup-Maker/blob/master/classes/Utils/Alerts.php#L437-L483